random_walker.py

```diff
@@ -0,0 +1,221 @@
+        """
+        Calculates the step length for a given direction.
+        Override in subclass in order to implement an other behaviour
+
+        :param walk_direction: direction for this step
+        :return: step length
+        """
+        return self.step_length
+
+    def reset_position(self, index: int):
+        """
+        Reset the current position from index to the previous position
+
+        :param index: current step
+        :return: None
+        """
+        self.x_positions[index] = self.x_positions[index - 1]
+        self.y_positions[index] = self.y_positions[index - 1]
+
+    def calculate_position(self, index: int, playground: Playground):
+        """
+        Calculate the next position by randomly select a new walk direction
+        and moving self.get_walk_step_length() in this direction.
+        The position will be reset to the previous one if the position is outside the playground
+
+        :param index: step index
+        :param playground: Playground for the walker
+        :return: None
+        """
+        walk_direction = self.get_random_walk_direction()
+        self.x_positions[index] = \
+            self.x_positions[index - 1] + \
+            walk_direction.get_x_factor() * self.get_walk_step_length(walk_direction)
+        self.y_positions[index] = \
+            self.y_positions[index - 1] + \
+            walk_direction.get_y_factor() * self.get_walk_step_length(walk_direction)
+        if not playground.is_position_in_playground(self.x_positions[index],
+                                                    self.y_positions[index]):
+            self.reset_position(index)
+
+    def execute_random_walk(self, playground: Playground, start_x: int = 0, start_y: int = 0):
+        """
+        Execute the random walk calculation for all steps, starting at start_x,
+        start_y on a playground
+
+        :param playground: Playground for the walker
+        :param start_x: start position x
+        :param start_y: start position y
+        :return: None
+        """
+        self.x_positions[0] = start_x
+        self.y_positions[0] = start_y
+        for index in range(1, self.steps):
+            self.calculate_position(index, playground)
+
+
+def create_different_walkers(count: int, steps: int) -> List[RandomWalker]:
+    """
+    Walker Factory creating random walkers which were inherited from the class RandomWalker
+
+    :param count: number of walkers
+    :param steps: steps for the walkers
+    :return: Array of Walkers
+    """
+    result = []
+    # get subclasses from RandomWalker
+    types = RandomWalker.__subclasses__()
+    for i in range(count):
+        # pick random subclass
+        random_type = types[random.randint(0, len(types) - 1)]
+        # generate object
+        random_object = random_type(steps)
+        # add object to result array
+        result.append(random_object)
+    return result
+
+
+class Rook(RandomWalker):
```

Comment:
nice cascading of walker classes and using chess names / characteristics is quite creative